File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/klaytnDebug/profiling/DebugBlockProfileTest.java
package opensdk.sdk.apis.klaytnDebug.profiling;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.DebugBlockProfileResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Debug RPC Test")
public class DebugBlockProfileTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC debug_blockProfile")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        String file = "block.profile";
        int seconds = 4;

        DebugBlockProfileResponse response = sdk.debug.blockProfile(file, seconds).send();
        response.getResult();
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/debug/profiling/DebugBlockProfile.test.js
const OpenSdk = require("opensdk-javascript");
const { expect } = require("@jest/globals");
const { RPC } = require("../../constant");

const sdk = new OpenSdk(new OpenSdk.ApiClient(RPC));

describe('debug_blockProfile API', () => {
    test('should return debug_blockProfile', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined();
            expect(data).toBe(null);
            done();
        };

        const file = "block.profile";
        const seconds = 10;

        sdk.debug.blockProfile(file, seconds, {}, callbackOne);
    });
});

